tests/test_constrain_current.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+import desc.io
+from desc.compute import data_index
+from desc.compute._core import compute_rotational_transform
+from desc.compute.utils import compress
+from desc.grid import QuadratureGrid, ConcentricGrid, LinearGrid
+from desc.transform import Transform
+
+
+class _ExactValueProfile:
+    """Monkey patches the compute method of desc.Profile for testing."""
+
+    def __init__(self, eq):
+        self.eq = eq
+
+    def compute(self, params=None, grid=None, dr=0, *args):
+        # unused arguments required to override profile.compute
+        if dr == 0:
+            return self.eq.compute("current", grid)["current"]
+        if dr == 1:
+            return self.eq.compute("current_r", grid)["current_r"]
+
+
+class TestConstrainCurrent:
+    """Tests for running DESC with a fixed current profile."""
+
+    def test_compute_rotational_transform(self, DSHAPE, HELIOTRON):
```

Comment:
This would fit under `test_compute_funs.py`. That file seemed bloated. It has global variables for Mercier and convolution stuff; I thought adding the `_ExactValueProfile` class there would make it disorganized.

If I should move it there let me know.

Reply:
I think it's fine as its own file, there are a ton of compute functions so it makes sense to start breaking things up.